girder/utility/server.py

```diff
@@ -176,8 +176,8 @@ def setup(test=False, plugins=None, curConfig=None):
         if name != constants.GIRDER_ROUTE_ID and name in pluginWebroots:
             cherrypy.tree.mount(pluginWebroots[name], route, appconf)
 
-    if test:
-        application.merge({'server': {'mode': 'testing'}})
+    if (mode and mode != ServerMode.PRODUCTION):
```

Comment:
Do we need the `mode != ServerMode.PRODUCTION` check? If `setup()` is passed an affirmative `mode`, should we just put it into the `application` config?

More significantly, is this needed here at all? Isn't `configureServer` (which is always called) already doing the same thing (updating `curConfig['server']['mode']` with a passed `mode`)?

Reply:
I agree, it doesn't look like it's needed, and removing it has no noticeable effect.